src/types.rs

```diff
@@ -111,3 +111,58 @@ impl TryFrom<u8> for Channels {
         }
     }
 }
+
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum RawChannels {
+    Rgb,
+    Bgr,
+    Rgba,
+    Argb,
+    Rgbx,
+    Xrgb,
+    Bgra,
+    Abgr,
+    Bgrx,
+    Xbgr,
+}
+
+impl From<Channels> for RawChannels {
+    fn from(value: Channels) -> Self {
+        match value {
+            Channels::Rgb => Self::Rgb,
+            Channels::Rgba => Self::Rgba,
+        }
+    }
+}
+
+impl From<RawChannels> for Channels {
```

Comment:
Minor nit. This is a lossy conversion, `From<...>` is usually discouraged in such cases, perhaps `.image_channels()`, `.target_channels()` or something like this would be more appropriate?